packages/graphql/src/translate/projection/elements/create-projection-subquery.ts

```diff
@@ -0,0 +1,182 @@
+import { AUTH_FORBIDDEN_ERROR } from "../../../constants";
+import { toNumber } from "../../../utils/utils";
+
+export function createProjectionSubquery({
+    parentNode,
+    whereInput,
+    node,
+    context,
+    alias,
+    nestedProjection,
+    nestedSubqueries,
+    relationField,
+    relationshipDirection,
+    optionsInput,
+    authValidateStrs,
+}: {
+    parentNode: CypherBuilder.Node;
+    whereInput?: GraphQLWhereArg;
+    node: Node;
+    context: Context;
+    alias: string; // TODO: this should be output variable instead
+    nestedProjection: string;
+    nestedSubqueries: CypherBuilder.Clause[];
+    relationField: RelationField;
+    relationshipDirection: RelationshipDirection;
+    optionsInput: GraphQLOptionsArg;
+    authValidateStrs: string[] | undefined;
+}): {
+    subquery: CypherBuilder.Clause;
+    projectionColumn: CypherBuilder.ProjectionColumn[];
+} {
+    const isArray = relationField.typeMeta.array;
+    const targetNode = new CypherBuilder.NamedNode(alias, {
+        labels: node.getLabels(context),
+    });
+
+    const relationship = new CypherBuilder.Relationship({
+        source: parentNode,
+        target: targetNode,
+        type: relationField.type,
+    });
+    if (relationshipDirection === "IN") {
+        relationship.reverse();
+    }
+
+    const isUndirected = relationshipDirection === "undirected";
+    const pattern = new CypherBuilder.Pattern(relationship, { directed: !isUndirected });
+
+    const subqueryMatch = new CypherBuilder.Match(pattern);
+
+    const projection = new CypherBuilder.RawCypher((env) => {
+        // TODO: use MapProjection
+        return `${targetNode.getCypher(env)} ${nestedProjection}`;
+    });
+
+    if (whereInput) {
+        const wherePredicate = createCypherWherePredicate({
+            element: node,
+            context,
+            whereInput,
+            targetElement: targetNode,
+        });
+        if (wherePredicate) subqueryMatch.where(wherePredicate); // TODO: should return var a projectionColumn
+    }
+
+    const whereAuth = createAuthPredicates({
+        entity: node,
+        operations: "READ",
+        context,
+        where: {
+            varName: alias,
+            node,
+        },
+    });
+
+    if (whereAuth) {
+        subqueryMatch.and(whereAuth);
+    }
+
+    const preAuth = createAuthPredicates({
+        entity: node,
+        operations: "READ",
+        context,
+        allow: {
+            parentNode: node,
+            varName: alias,
+        },
+    });
+
+    if (preAuth) {
+        const allowAuth = new CypherBuilder.apoc.ValidatePredicate(CypherBuilder.not(preAuth), AUTH_FORBIDDEN_ERROR);
+        subqueryMatch.and(allowAuth);
+    }
+
+    if (authValidateStrs?.length) {
+        const authValidateStatements = authValidateStrs.map((str) => new CypherBuilder.RawCypher(str));
+        const authValidatePredicate = CypherBuilder.and(...authValidateStatements) as CypherBuilder.Predicate;
```

Comment:
Is there a reason `CypherBuilder.and()` isn't able to return the type `CypherBuilder.Predicate`?

Reply:
`CypherBuilder.and` returns `Predicate | undefined` as the typings system is not clever enough to detect that there are elements in the `authValidateStatements` array (checked by the previous if)